staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanagertest/testfieldmanager.go

```diff
@@ -0,0 +1,224 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanagertest
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager"
+	"k8s.io/kube-openapi/pkg/util/proto"
+	prototesting "k8s.io/kube-openapi/pkg/util/proto/testing"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/v4/merge"
+	"sigs.k8s.io/structured-merge-diff/v4/typed"
+)
+
+var kubernetesSwaggerSchema = prototesting.Fake{
+	Path: filepath.Join(
+		strings.Repeat(".."+string(filepath.Separator), 8),
+		"api", "openapi-spec", "swagger.json"),
+}
+
+// NewBuiltinTypeConverter creates a TypeConverter with all the built-in
+// types defined, given the committed kubernetes swagger.json.
+func NewBuiltinTypeConverter() fieldmanager.TypeConverter {
+	tc, err := fieldmanager.NewTypeConverter(newFakeOpenAPIModels(), false)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to build TypeConverter: %v", err))
+	}
+	return tc
+}
+
+func newFakeOpenAPIModels() proto.Models {
+	d, err := kubernetesSwaggerSchema.OpenAPISchema()
+	if err != nil {
+		panic(err)
+	}
+	m, err := proto.NewOpenAPIData(d)
+	if err != nil {
+		panic(err)
+	}
+	return m
+}
+
+type fakeObjectConvertor struct {
+	converter  merge.Converter
+	apiVersion fieldpath.APIVersion
+}
+
+//nolint:staticcheck,ineffassign // SA4009 backwards compatibility
+func (c *fakeObjectConvertor) Convert(in, out, context interface{}) error {
+	if typedValue, ok := in.(*typed.TypedValue); ok {
+		var err error
+		out, err = c.converter.Convert(typedValue, c.apiVersion)
+		return err
+	}
+	return nil
+}
+
+func (c *fakeObjectConvertor) ConvertToVersion(in runtime.Object, _ runtime.GroupVersioner) (runtime.Object, error) {
+	return in, nil
+}
+
+func (c *fakeObjectConvertor) ConvertFieldLabel(_ schema.GroupVersionKind, _, _ string) (string, string, error) {
+	return "", "", errors.New("not implemented")
+}
+
+type fakeObjectDefaulter struct{}
+
+func (d *fakeObjectDefaulter) Default(in runtime.Object) {}
+
+type sameVersionConverter struct{}
+
+func (sameVersionConverter) Convert(object *typed.TypedValue, version fieldpath.APIVersion) (*typed.TypedValue, error) {
+	return object, nil
+}
+
+func (sameVersionConverter) IsMissingVersionError(error) bool {
+	return false
+}
+
+// NewFakeObjectCreater implements ObjectCreater, it can create empty
+// objects (unstructured) of the given GVK.
+func NewFakeObjectCreater(gvk schema.GroupVersionKind) runtime.ObjectCreater {
+	return &fakeObjectCreater{gvk: gvk}
+}
+
+type fakeObjectCreater struct {
+	gvk schema.GroupVersionKind
+}
+
+func (f *fakeObjectCreater) New(_ schema.GroupVersionKind) (runtime.Object, error) {
```

Comment:
Why does this function take the GVK parameter if it's ignored?

Reply:
OK, I never took the time to change this since it just worked and changing that would break some tests, but I figured out why the test is broken. I'll fix this, thanks :-)